### PR TITLE
[KUBOS-493] Updating HAL docs

### PR DESCRIPTION
**Pull request by plauche**
* Filling out api level documentation
* Adding code examples to function docs
* Code formatting

### CODE REVIEW

File: hal/kubos-hal/kubos-hal/sdio.h [hunk 1]
```diff
@@ -39,39 +39,109 @@ typedef enum
...
+ * @param count [in] number of blocks to write
+ * @return KSDIOStatus SDIO_OK if successful, otherwise error code
+ */
 KSDIOStatus k_sdio_write_blocks(uint32_t * buffer, uint64_t addr, uint32_t block_size, uint32_t count);
 
+/**
+ * Reads blocks of data via SDIO
+ * @param buffer [out] data buffer to read data into
+ * @param addr [in] address on sdcard to read from
+ * @param block_size [in] size of each block to write
```
Comment (CatFreed) [inline hal/kubos-hal/kubos-hal/sdio.h]: nit: 'to read'

File: hal/kubos-hal/kubos-hal/spi.h [hunk 2]
```diff
@@ -197,35 +208,140 @@ typedef struct {
...
     csp_mutex_t spi_lock;
 } KSPI;
 
 /**
- * Setup and enable spi bus
- * @param spi spi bus to initialize
+ * @brief Configures and enables a SPI bus
+ *
+ * This function is used to configure and enable SPI buses for further usage (reading/writing).
+ * Calling this function is always the first step before using any SPI perpherial. This function
```
Comment (CatFreed) [inline hal/kubos-hal/kubos-hal/spi.h]: nit: spelling 'peripheral'

File: hal/kubos-hal/kubos-hal/spi.h [hunk 3]
```diff
@@ -197,35 +208,140 @@ typedef struct {
...
+    .clock_polarity = K_SPI_CPOL_HIGH,
+    .clock_phase = K_SPI_CPHA_1EDGE,
+    .first_bit = K_SPI_FIRSTBIT_LSB,
+    .speed = 100000
+};
+
+k_spi_init(K_SPI1, &conf);
+ * @endcode
+ *
+ * @note This function delegates the low level initialization to the platform specific @ref kprv_spi_dev_init
```
Comment (CatFreed) [inline hal/kubos-hal/kubos-hal/spi.h]: nit: hyphenate 'low-level'

File: hal/kubos-hal/kubos-hal/spi.h [hunk 4]
```diff
@@ -197,35 +208,140 @@ typedef struct {
...
+ * @param spi number spi bus to initialize
  * @param conf config values to initialize with
  */
 void k_spi_init(KSPINum spi, KSPIConf * conf);
 
 /**
- * Terminate spi bus
+ * @brief Terminates a SPI bus
+ *
+ * This function is used to terminate a active SPI bus. After calling this function, the bus number
```
Comment (CatFreed) [inline hal/kubos-hal/kubos-hal/spi.h]: nit: 'an active'

File: hal/kubos-hal/kubos-hal/uart.h [hunk 5]
```diff
@@ -85,7 +84,7 @@
  * Available uart interfaces
  */
 typedef enum {
-    // @warning - need to add K_UART_NO_BUS
+// @warning - need to add K_UART_NO_BUS
```
Comment (CatFreed) [inline hal/kubos-hal/kubos-hal/uart.h]: Is there a JIRA story for this?

File: hal/kubos-hal/kubos-hal/uart.h [hunk 6]
```diff
@@ -146,11 +145,12 @@ typedef enum {
...
  * Uart configuration structure
  */
-typedef struct {
+typedef struct
+{
     /**
      * The path of the uart bus
      */
-    const char *dev_path;
+    const char * dev_path;
```
Comment (CatFreed) [inline hal/kubos-hal/kubos-hal/uart.h]: I have the PointerAlignment setting in my .clang-format set to 'Right'. Do we want to make 'Middle' the standard?

File: hal/kubos-hal/kubos-hal/uart.h [hunk 7]
```diff
@@ -188,10 +188,23 @@ typedef struct {
...
+    /**
+     * UART device number
+     */
     int dev;
+    /**
+     * Copy of UART configuration options
+     */
     KUARTConf conf;
+    /**
+     * Queue filled with received uart data
```
Comment (CatFreed) [inline hal/kubos-hal/kubos-hal/uart.h]: nit: capitalize UART

File: hal/kubos-hal/kubos-hal/gpio.h [hunk 8]
```diff
@@ -16,7 +16,7 @@
  */
 
 /**
- * @defgroup GPIO
+ * @defgroup GPIO HAL GPIO Interface
```
Comment (plauche) [inline hal/kubos-hal/kubos-hal/gpio.h]: The first GPIO is the name/reference of the group, "HAL GPIO Interface" is the displayed title of the group.

File: hal/kubos-hal/kubos-hal/uart.h (hunk 5 above)
Comment (plauche) [inline hal/kubos-hal/kubos-hal/uart.h]: There is not....and now I can't remember why I put that comment there...

File: hal/kubos-hal/kubos-hal/gpio.h (hunk 8 above)
Comment (CatFreed) [inline hal/kubos-hal/kubos-hal/gpio.h]: :+1:

File: hal/kubos-hal/kubos-hal/uart.h (hunk 5 above)
Comment (CatFreed) [inline hal/kubos-hal/kubos-hal/uart.h]: :laughing: There's probably a compiler error or something if you try to use UART, but your project config says that your target has no UART buses

File: hal/kubos-hal/kubos-hal/uart.h (hunk 6 above)
Comment (plauche) [inline hal/kubos-hal/kubos-hal/uart.h]: I prefer the look of 'Middle' and that is what the .clang-format was set to when this file was formatted :smile:

File: hal/kubos-hal/kubos-hal/i2c.h [hunk 9]
```diff
@@ -170,7 +170,10 @@ typedef enum {
  k_i2c_init(K_I2C1, &conf);
  * @endcode
  *
- * @param i2c i2c bus to initialize
+ * @note The functions k_i2c_default_init or k_i2c_default_dev_init can also be used to initialize an
+ * I2C device. They provide nice convience wrappers around k_i2c_init.
```
Comment (CatFreed) [inline hal/kubos-hal/kubos-hal/i2c.h]: nit: spelling 'convenience'

File: docs/kubos-hal/index.rst [hunk 10]
```diff
@@ -1,6 +1,8 @@
 Kubos HAL Documentation
 =======================
 
+.. warning:: The Kubos-HAL functionality is currently only available on Kubos-RT.
```
Comment (kyleparrott) [inline docs/kubos-hal/index.rst]: Should it be `Kubos HAL` and `Kubos RT`?

**Pull request merged**